hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockOutputStream.java

```diff
@@ -146,6 +149,8 @@ public class BlockOutputStream extends OutputStream {
   private final ContainerClientMetrics clientMetrics;
   private boolean allowPutBlockPiggybacking;
 
+  private volatile CompletableFuture<Void> lastFlushFuture;
```

Comment:
`volatile` won't be necessary in this case because the access to this variable is already in `synchronized` blocks.

```suggestion
  private CompletableFuture<Void> lastFlushFuture;
```

Reply:
Good point. Updated.